pkg/ddevapp/addons.go

```diff
@@ -166,6 +224,477 @@ func ProcessAddonAction(action string, dict map[string]interface{}, bashPath str
+	err = createConfigurationFiles(app)
+	if err != nil {
+		return fmt.Errorf("failed to create configuration files for PHP action: %w", err)
+	}
+
+	// Create a shell script that validates original PHP syntax first, then executes with strict mode
+	shellScript := fmt.Sprintf(phpActionShellScriptTemplate, originalAction, action)
+
+	cmd := []string{"sh", "-c", shellScript}
+
+	// Bind mount the .ddev directory and the project root into the container
+	binds := []string{fmt.Sprintf("%s:/var/www/html", app.AppRoot)}
+
+	// Build environment variables array with standard DDEV variables
+	env := buildPHPActionEnvironment(app, verbose)
+
+	// Create in-memory docker-compose project for PHP action execution
+	phpProject, err := dockerutil.CreateComposeProject("name: ddev-php-action")
+	if err != nil {
+		return fmt.Errorf("failed to create compose project: %v", err)
+	}
+
+	// Create service configuration for PHP action
+	serviceName := "php-runner"
+	serviceConfig := composeTypes.ServiceConfig{
+		Name:        serviceName,
+		Image:       image,
+		Command:     cmd,
+		WorkingDir:  "/var/www/html/.ddev",
+		Environment: buildEnvironmentMap(env),
+		Volumes:     buildVolumeConfigs(binds),
+		Networks: map[string]*composeTypes.ServiceNetworkConfig{
+			"default":          {},
+			dockerutil.NetName: {},
+		},
+		Labels: composeTypes.Labels{
+			"com.ddev.site-name": app.Name + "-php-action",
+			"com.ddev.approot":   app.AppRoot,
+		},
+	}
+
+	// Add host.docker.internal setup using DDEV's standard logic
+	hostDockerInternalIP, err := dockerutil.GetHostDockerInternalIP()
+	if err != nil {
+		util.Warning("Could not determine host.docker.internal IP address: %v", err)
+	}
+
+	// Set up host.docker.internal based on DDEV's standard approach
+	if hostDockerInternalIP != "" {
+		// Use specific IP address for host.docker.internal
+		extraHosts, err := composeTypes.NewHostsList([]string{
+			"host.docker.internal:" + hostDockerInternalIP,
+		})
+		if err == nil {
+			serviceConfig.ExtraHosts = extraHosts
+		}
+	} else if (runtime.GOOS == "linux" && !nodeps.IsWSL2() && !dockerutil.IsColima()) ||
+		(nodeps.IsWSL2() && globalconfig.DdevGlobalConfig.XdebugIDELocation == globalconfig.XdebugIDELocationWSL2) {
+		// Use host-gateway for modern Docker on Linux
+		extraHosts, err := composeTypes.NewHostsList([]string{
+			"host.docker.internal:host-gateway",
+		})
+		if err == nil {
+			serviceConfig.ExtraHosts = extraHosts
+		}
+	}
+
+	phpProject.Services[serviceName] = serviceConfig
+
+	// Add network definitions matching DDEV's standard networking
+	if phpProject.Networks == nil {
+		phpProject.Networks = composeTypes.Networks{}
+	}
+	phpProject.Networks[dockerutil.NetName] = composeTypes.NetworkConfig{
+		Name:     dockerutil.NetName,
+		External: true,
+	}
+	phpProject.Networks["default"] = composeTypes.NetworkConfig{
+		Name: "php-action-" + app.GetComposeProjectName() + "_default",
+	}
```

Comment:
Add labels to delete network on `dev poweroff` and delete the network itself here:

```suggestion
	phpProject.Networks["default"] = composeTypes.NetworkConfig{
		Name:   "php-action-" + app.GetComposeProjectName() + "_default",
		Labels: composeTypes.Labels{"com.ddev.platform": "ddev"},
	}
	defer func() {
		dockerutil.RemoveNetworkWithWarningOnError(phpProject.Networks["default"].Name)
	}()
```